src/app/components/common/node-editor/node-editor.component.ts

```diff
@@ -79,6 +79,18 @@ export class NodeEditorComponent implements OnInit {
   isAddFilterWizard: boolean = false;
   isAlive: boolean;
 
+  notificationChangedConfig: any;
+  rulePluginChangedConfig: any;
+  ruleCategory: any;
+  deliveryCategory: any;
+  deliveryPluginChangedConfig: any;
+  ruleConfiguration: any;
+  deliveryConfiguration: any;
+  showConfigureModal = false;
+  isServiceAvailable = false;
+  serviceInfo: {};
+  btnText = '';
```

Comment:
Can't we avoid those many variables for managing the notification instance. We should look for the better data structure to avoid all these variables.

Reply:
I've removed one variable i.e. `showConfigureModal`. All other variables seems required, which I can't remove.